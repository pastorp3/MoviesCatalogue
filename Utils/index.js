import { checkPropTypes } from 'prop-types';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import RootReducer from '../src/reducers/index';

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return component;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsErr;
}
