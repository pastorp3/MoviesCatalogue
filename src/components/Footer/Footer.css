.footer {
  background-color: blue;
  color: white;
  font-size: 18px;
  font-family: sans-serif;
  height: 70px;
  text-align: center;
  padding-top: 40px;
}
