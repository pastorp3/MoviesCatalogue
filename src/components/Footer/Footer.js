import './Footer.css';

const Footer = () => <div className="footer"><span>Copyright 2021</span></div>;

export default Footer;
