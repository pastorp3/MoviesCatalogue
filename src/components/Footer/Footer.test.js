/* eslint-disable  preact/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Footer from './Footer';
import { findByTestAtrr } from '../../../Utils';

const setUp = (props = {}) => {
  const component = shallow(<Footer {...props} />);
  return component;
};

describe('Footer Component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it('Should render whitout error', () => {
    const wrapper = findByTestAtrr(component, 'Footer');
    expect(wrapper.length).toBe(1);
  });

  it('Should render according with snapshout whithout error', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/* eslint-enable */
