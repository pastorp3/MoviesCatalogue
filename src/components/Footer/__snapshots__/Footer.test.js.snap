// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Should render according with snapshout whithout error 1`] = `
<div
  className="footer"
  data-test="Footer"
>
  <span>
    Copyright 2021
  </span>
</div>
`;
